src/backend/cdb/cdbmutate.c

```diff
@@ -2559,8 +2586,6 @@ shareinput_mutator_xslice_3(Node *node, PlannerInfo *root, bool fPop)
 
 		if (list_member_int(ctxt->qdShares, sisc->share_id))
 		{
-			Assert(sisc->scan.plan.flow);
-			Assert(sisc->scan.plan.flow->flotype == FLOW_SINGLETON);
```

Comment:
It seems that we are still able to validate that slice under modId is singleton.

Reply:
only if `sisc->scan.plan.flow != NULL`

Reply:
done